.travis.yml

```diff
@@ -0,0 +1,56 @@
+language: go
+
+services:
+  - docker
+
+go:
+  - "1.9.x"
+
+before_install:
+  - . $HOME/.nvm/nvm.sh
+  - nvm install 8
+  - nvm use 8
+  - npm install -g yarn
+
+install:
+  - make dependencies
+
+stages:
+  - name: validate and build
+    if: >-
+      type = pull_request OR
+      branch IN (master, staging) OR
+      tag IS present
+  - name: release to github
+    if: tag IS present
+  - name: push to dockerhub
+    if: tag IS present
+
+jobs:
+  include:
+    # validate and build
+    - stage: validate and build
+      script:
+        - make test
+        - make lint
+        - make coverage
+        - make validate-commit
+    - script:
+        - make packages
+    # release to github
+    - stage: release to github
+      script:
+        - make packages
+      deploy:
+        provider: releases
+        api_key: $GITHUB_TOKEN
+        file_glob: true
+        file: build/*.tar.gz
+        skip_cleanup: true
+        on:
+          all_branches: true
+    # push to dockerhub
+    - stage: push to dockerhub
+      script:
+        - make build
```

Comment:
We should avoid building binaries for all OSes and arches here, building only the one that we use for Docker.

Reply:
you're right @smola 

What if I change it with:
```yaml
script:
  - PKG_OS=linux make build
  - DOCKER_PUSH_LATEST=true make docker-push
```
(I realized that even removing the `make build` command, the `Darwin` binary will be also generated, so it seems to be needed the `PKG_OS=linux` env var, isn't it?)

We also need the `make build` command because there are "other things" that needs to be built for the docker image.